webdictionary/src/main/java/com/eis0/kademlianetwork/KademliaMessage.java

```diff
@@ -0,0 +1,105 @@
+package com.eis0.kademlianetwork;
+
+import com.eis.smslibrary.SMSMessage;
+import com.eis.smslibrary.SMSPeer;
+import com.eis0.kademlia.KademliaId;
+
+/**
+ * This class is used to create custom messages to send or receive inside the Kademlia Network, the
+ * fields are in common for all the type of requests
+ */
+public class KademliaMessage {
+    private static final String REQUEST_TYPE_NULL = "The requestType parameter is null";
+
+    public SMSPeer peer;
+    public String text;
+    public RequestTypes requestType;
+    public KademliaId idToFind;
+    public SMSPeer searcher;
+    public String key;
+    public String resource;
+
+    /**
+     * This is the constructor of the class, it allows to create an instance of the KademliaMessage
+     * with the content of the received {@link SMSMessage}; it automatically extracts the values in
+     * the fields of the message.
+     * The message must satisfy the division into fields, every single word inside of it has a
+     * specific task, except for the 'resource' field, which has an arbitrary length
+     * [0] => The {@link RequestTypes} of the message
+     * [1] => The {@link KademliaId}, can be the target ID to find inside the network, or the ID found
+     * [2] => The {@link SMSPeer} which originated the request
+     * [3] => The key of the interested resource
+     * [4 --> end of the message] => The resource brought by the message
+     *
+     * @param message The message received from the network
+     * @throws IndexOutOfBoundsException                                     If the message doesn't have the minimum number of fields
+     * @throws IllegalArgumentException                                      If the {@link KademliaId} has not the right fit
+     * @throws com.eis.smslibrary.exceptions.InvalidTelephoneNumberException If If telephoneNumber
+     *                                                                       check is not {@link SMSPeer.TelephoneNumberState#TELEPHONE_NUMBER_VALID}.
+     */
+    public KademliaMessage(SMSMessage message) {
```

Comment:
Since the "output" of this class is a String, maybe consider doing the same with the "input".
Just to be coherent, I would have a constructor `KademliaMessage(String message, Peer sender)`. Or a method `getMessage()` that returns a `SMSMessage ` built from the `toString()`.

Reply:
I think it would be even better to move the peer handling process outside the KademliaMessage class, because as a value it's easily obtainable; this way, I can let the class process only the text message, as you said, inserted ad a String value